tests/unit-core/test-promise-callback.c

```diff
@@ -0,0 +1,283 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript.h"
+
+#include "test-common.h"
+
+typedef enum
+{
+  C = JERRY_PROMISE_CALLBACK_CREATE, /**< same as JERRY_PROMISE_CALLBACK_CREATE with undefined value */
+  RS = JERRY_PROMISE_CALLBACK_RESOLVE, /**< same as JERRY_PROMISE_CALLBACK_RESOLVE */
+  RJ = JERRY_PROMISE_CALLBACK_REJECT, /**< same as JERRY_PROMISE_CALLBACK_REJECT */
+  BR = JERRY_PROMISE_CALLBACK_BEFORE_REACTION_JOB, /**< same as JERRY_PROMISE_CALLBACK_BEFORE_REACTION_JOB */
+  AR = JERRY_PROMISE_CALLBACK_AFTER_REACTION_JOB, /**< same as JERRY_PROMISE_CALLBACK_AFTER_REACTION_JOB */
+  A = JERRY_PROMISE_CALLBACK_ASYNC_AWAIT, /**< same as JERRY_PROMISE_CALLBACK_ASYNC_AWAIT */
+  BRS = JERRY_PROMISE_CALLBACK_ASYNC_BEFORE_RESOLVE, /**< same as JERRY_PROMISE_CALLBACK_ASYNC_BEFORE_RESOLVE */
+  BRJ = JERRY_PROMISE_CALLBACK_ASYNC_BEFORE_REJECT, /**< same as JERRY_PROMISE_CALLBACK_ASYNC_BEFORE_REJECT */
+  ARS = JERRY_PROMISE_CALLBACK_ASYNC_AFTER_RESOLVE, /**< same as JERRY_PROMISE_CALLBACK_ASYNC_AFTER_RESOLVE */
+  ARJ = JERRY_PROMISE_CALLBACK_ASYNC_AFTER_REJECT, /**< same as JERRY_PROMISE_CALLBACK_ASYNC_AFTER_REJECT */
+  CP = UINT8_MAX - 1, /**< same as JERRY_PROMISE_CALLBACK_CREATE with Promise value */
+  E = UINT8_MAX, /**< marks the end of the event list */
+} jerry_promise_callback_event_abbreviations_t;
+
+static int user;
+static const uint8_t *next_event_p;
+
+static void
+promise_callback (jerry_promise_callback_notification_types_t notification_type, /**< notification type */
+                  const jerry_value_t object, /**< target object */
+                  const jerry_value_t value, /**< optional argument */
+                  void *user_p) /**< user pointer passed to the callback */
+{
+  TEST_ASSERT (user_p == (void *) &user);
+
+  switch (notification_type)
```

Comment:
~~Would it make sense to test if the notifications are triggered in the correct order?~~ I see they are already tested.